### DISCUSSION TITLE
Does someone have a good config for indent-blankline

### DISCUSSION THREAD

**Discussion by CaffeineB34r** [NvChad/NvChad#2780] [Q&A] [answered]
I'm  looking to do something like this
![265403990-67707d8e-57d3-411c-8418-77908d8babd9](https://github.com/NvChad/NvChad/assets/62663698/e1699038-3725-46b9-a5b5-bf9160529906)
iv'e been trying for a couple hours trying to write my own but i cant seem to get it to work

**Comment by ashb** [comment]
Try this in your chadrc @CaffeineB34r 

```lua
M.ui = {
  theme = "onedark",

  hl_override = {
    IblScopeChar = { fg = "#00FF00", nocombine = true },
  }
}
```

**Reply by siduck** [reply]
@ashb although its better to add that hlgroup in hl_override so he can use nvchad theme's colors

**Reply by CaffeineB34r** [reply]
the hl_override hint was really what i needed to simplify the plugin setup, i ended up using the hl_add to add the rainbow highlighting and will be posting my config in case anyone searches for something similar, thank you very much @ashb

**Comment by CaffeineB34r** [comment, marked as answer]
after some tinkering i ended with this in my custom plugins file 
plugins.lua:
```lua
  {
    "HiPhish/rainbow-delimiters.nvim",
    opts = {
      highlight = require "custom.configs.highlight",
    },
    config = function(_, opts)
      require("rainbow-delimiters.setup").setup(opts)
    end,
  },
  {
    "lukas-reineke/indent-blankline.nvim",
    version = false,
    dependencies = { "rainbow-delimiters.nvim" },
    config = function(_, opts)
      local hooks = require "ibl.hooks"
      hooks.register(hooks.type.SCOPE_HIGHLIGHT, hooks.builtin.scope_highlight_from_extmark)
      require("ibl").setup(opts)
    end,
    opts = function() --has to be a function to override the default nvchad options
      return {
        scope = {
          enabled = true,
          show_start = true,
          highlight = require "custom.configs.highlight",
        },
      }
    end,
  },
```
chadrc.lua
```lua
M.ui = {
  hl_add = {
    CatppuccinRed = { fg = "#e78284" },
    CatppuccinPeach = { fg = "#ef9f76" },
    CatppuccinYellow = { fg = "#e5c890" },
    CatppuccinGreen = { fg = "#a6d189" },
    CatppuccinBlue = { fg = "#85c1dc" },
    CatppuccinPurple = { fg = "#babbf1" },
  },
}
```


it updates the plugin and sets up rainbow delimiters with scope highlighting
![Screenshot from 2024-05-01 18-01-07](https://github.com/NvChad/NvChad/assets/62663698/95932c07-12b6-4a4b-b45a-a71f8f74a329)
